src/com/facebook/buck/ide/intellij/IjModuleGraphFactory.java

```diff
@@ -315,6 +325,36 @@ public static IjModuleGraph from(
     return new IjModuleGraph(depsBuilder.build());
   }
 
+  private static IjModule createExtraModuleForCompilerOutput(
+      IjModule module, IjProjectConfig projectConfig, ProjectFilesystem projectFilesystem) {
+    Path projectRootPath = Paths.get(projectConfig.getProjectRoot());
+    Path extraModuleBasePath = projectRootPath
+        .resolve("buck-out")
+        .resolve("extra-intellij-output")
+        .resolve(projectRootPath.relativize(module.getModuleBasePath()));
+
+    Path extraModuleRelativePath = projectRootPath.relativize(extraModuleBasePath.normalize());
+    if (!projectFilesystem.exists(extraModuleRelativePath)) {
+      try {
+        projectFilesystem.mkdirs(extraModuleRelativePath);
```

Comment:
When generating a real project (I used uber/okbuck sample app), without this logic, I get the exception below. It looks like the `FakeProjectFilesystem` used in the test code does not throw the same exception since it's `.exists()` method only checks whether its a file or directory and not the presence of the actual file on disk
```
[2017-08-11 11:56:17.363][error][command:null][tid:106][com.facebook.buck.cli.Main] /Users/jbarr/Uber/okbuck/buck-out/extra-intellij-output/app
java.nio.file.NoSuchFileException: /Users/jbarr/Uber/okbuck/buck-out/extra-intellij-output/app
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
	at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144)
	at java.nio.file.Files.readAttributes(Files.java:1737)
	at com.facebook.buck.io.ProjectFilesystem$FileTreeWalker.getAttributes(ProjectFilesystem.java:1190)
	at com.facebook.buck.io.ProjectFilesystem$FileTreeWalker.walk(ProjectFilesystem.java:1107)
	at com.facebook.buck.io.ProjectFilesystem$FileTreeWalker.access$2(ProjectFilesystem.java:1106)
	at com.facebook.buck.io.ProjectFilesystem.walkFileTree(ProjectFilesystem.java:535)
	at com.facebook.buck.io.ProjectFilesystem.walkRelativeFileTree(ProjectFilesystem.java:524)
	at com.facebook.buck.io.ProjectFilesystem.walkRelativeFileTree(ProjectFilesystem.java:481)
	at com.facebook.buck.ide.intellij.IjProjectTemplateDataPreparer.createExcludes(IjProjectTemplateDataPreparer.java:228)
	at com.facebook.buck.ide.intellij.IjProjectTemplateDataPreparer.getContentRoots(IjProjectTemplateDataPreparer.java:291)
	at com.facebook.buck.ide.intellij.IjProjectWriter.write(IjProjectWriter.java:59)
	at com.facebook.buck.ide.intellij.IjProject.write(IjProject.java:113)
	at com.facebook.buck.ide.intellij.IjProjectCommandHelper.writeProjectAndGetRequiredBuildTargets(IjProjectCommandHelper.java:269)
	at com.facebook.buck.ide.intellij.IjProjectCommandHelper.runIntellijProjectGenerator(IjProjectCommandHelper.java:233)
	at com.facebook.buck.ide.intellij.IjProjectCommandHelper.parseTargetsAndRunProjectGenerator(IjProjectCommandHelper.java:196)
	at com.facebook.buck.cli.ProjectCommand.runWithoutHelp(ProjectCommand.java:317)
	at com.facebook.buck.cli.AbstractCommand.run(AbstractCommand.java:227)
	at com.facebook.buck.cli.AbstractContainerCommand.run(AbstractContainerCommand.java:67)
	at com.facebook.buck.cli.BuckCommand.run(BuckCommand.java:81)
	at com.facebook.buck.cli.Main.runMainWithExitCode(Main.java:983)
	at com.facebook.buck.cli.Main.runMainThenExit(Main.java:372)
	at com.facebook.buck.cli.Main.nailMain(Main.java:1725)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.martiansoftware.nailgun.NGSession.run(NGSession.java:329)
```